wrappers/binaries.go

```diff
@@ -21,13 +21,57 @@
 package wrappers
 
 import (
+	"io"
 	"os"
+	"regexp"
+	"strconv"
+	"strings"
 
 	"github.com/snapcore/snapd/dirs"
 	"github.com/snapcore/snapd/osutil"
 	"github.com/snapcore/snapd/snap"
 )
 
+func detectLegacy() (noCompletion bool, legacyCompletion bool) {
+	fd, err := os.Open(dirs.BashCompletionScript)
+	if err != nil {
+		return true, false
+	}
+	defer fd.Close()
+	buf := new(strings.Builder)
+	io.Copy(buf, fd)
+	bash_completion := buf.String()
+
+	re := regexp.MustCompile(`(?:^|\n)# *RELEASE: ([0-9]+)[.]([0-9]+)(?:$|\n)`)
```

Comment:
I don't see `RELASE:` in that file on Arch, was is present in some older releases?

Reply:
Do you know which in bash version it was replaced with BASH_COMPLETION_VERSINFO? maybe it's so old that we don't need to care anymore?

Reply:
2.6. Xenial has 2.1 so we need to match `RELEASE:`. If it has `BASH_COMPLETION_VERSINFO` then it supports `XDG_DATA_DIRS` since that support was added in 2.2 which is prior to using `BASH_COMPLETION_VERSINFO`.

Reply:
Also it is bash-completion. The version of bash is irrelevant.

Reply:
I had changed the code to search patterns line by line like you suggested. This makes the order of checks irrelevant for performance.